ports/rp2/main.c

```diff
@@ -213,6 +226,9 @@ int main(int argc, char **argv) {
         #endif
         gc_sweep_all();
         mp_deinit();
+
+        // Hook for resetting anything after the MicroPy runtime has been deinitialised
+        MICROPY_BOARD_LATE_RESET();
```

Comment:
One more of the same. The equivalent hook for stm32 and renesas-ra is `MICROPY_BOARD_END_SOFT_RESET(&state)`, and it's placed before `gc_sweep_all()`. Would that work for your use case, also?

Reply:
I do not actively use this hook, so am happy to move it to before gc_sweep_all().

Though, I wonder if there are use-cases where you would want a call to occur post `mp_deinit()` 🤔. I assume your team chose its location before the gc for a reason.

Reply:
> Though, I wonder if there are use-cases where you would want a call to occur post mp_deinit() 🤔. I assume your team chose its location before the gc for a reason.

Good questions... I don't know the historical reason these hooks were added (there is some discussion in #6505 but it doesn't cover that.)

That said, I believe the reason to do it before `gc_sweep_all()` is to give you a chance to NULL out any pointers into the Python heap. If you have, for example, a C based interrupt handler that updates some data in the Python heap then that needs to be disabled and the pointer NULLed out before `gc_sweep_all()` blows away all of the Python heap contents. Otherwise it's going to be left with a dangling pointer to uninitialized Python heap space, and there may be a race condition where an interrupt fires and accesses that memory.

> I do not actively use this hook, so am happy to move it to before gc_sweep_all().

Sounds good, thanks. I'm wondering about the relative benefits of trimming back the hooks in this PR to just the ones you are using. Although I'm leaning towards the best outcome being if the hooks in rp2 match the ones in stm32/renesas, so it's worth adding them all if it achieves that goal.